features/customers.feature

```diff
@@ -220,3 +220,114 @@ Scenario: Deactivate and Activate a Customer
+    And I set the "Street" to "10th Casselberry Way"
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "This doesn't appear to be a valid email address" in the "Email" error string
+
+Scenario: Update a Customer with Missing First Name
+    When I visit the "Home Page"
+    And I press the "Search" button
+    Then I should see the message "Success"
+    When I set the "First Name" to " "
+    And I set the "Last Name" to "Test LN"
+    And I set the "Email" to "test_ff@domain.com"
+    And I select "True" in the "Active" dropdown
+    And I set the "street" to "10th Casselberry Way"
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "Required field" in the "First Name" error string
+    When I set the "First Name" to "test"
+    And I press the "Update" button
+    Then The "First Name" error string should be gone
+
+Scenario: Update a Customer with Missing Last Name
+    When I visit the "Home Page"
+    And I press the "Search" button
+    Then I should see the message "Success"
+    When I set the "First Name" to "Test FN"
+    And I set the "Last Name" to " "
+    And I set the "Email" to "test_ff@domain.com"
+    And I select "True" in the "Active" dropdown
+    And I set the "Street" to "10th Casselberry Way"
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "Required field" in the "Last Name" error string
+    When I set the "Last Name" to "test"
+    And I press the "Update" button
+    Then The "Last Name" error string should be gone
+
+Scenario: Update a Customer with Missing Password
+    When I visit the "Home Page"
+    And I press the "Search" button
+    Then I should see the message "Success"
+    When I set the "First Name" to "Test FN"
+    And I set the "Last Name" to "Test LN"
+    And I set the "Email" to "test_ff@domain.com"
+    And I set the "Password" to " "
+    And I select "True" in the "Active" dropdown
+    And I set the "Street" to "10th Casselberry Way"
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "Required field" in the "Password" error string
+    When I set the "Password" to "test"
+    And I press the "Update" button
+    Then The "Password" error string should be gone 
```

Comment:
Can we add the scenario for Updating with missing email - for completion.

Reply:
Done